src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -107,6 +112,13 @@ def on_train_start(self, trainer: "pl.Trainer", servable_module: "pl.LightningMo
         self.resp = requests.post(f"http://{self.host}:{self.port}/serve", json=payload)
         process.kill()
 
+        if is_overridden("configure_response", servable_module, ServableModule):
+            response = servable_module.configure_response()
+            assert self.resp.json() == response
+
+        if self.exit_on_failure and not self.successful:
+            raise SystemExit("The ServableModel API hasn't been properly implemented.")
```

Comment:
that is one of the few cases where IMO a MisconfigurationException would fit :)

Reply:
"hasn't been properly implemented" is unclear for users. Maybe direct them to a specific resource?